### ISSUE TITLE
FORK IT ALL if you fork it

### ISSUE THREAD

**Issue by rawpixel-vincent** [opensearch-project/OpenSearch#1400]
**Is your feature request related to a problem? Please describe.**
current aws client is just an http client.

**Describe the solution you'd like**
Something that doesn't makes me mad about aws forking elastic search for the wrong reason.

**Describe alternatives you've considered**
fork the elastic search clients and maintain them for open search so we can switch without changing our code.

**Additional context**
I'm an aws customer how setup elasticsearch not opensearch

**Comment by rawpixel-vincent** [comment]
I'm sorry do I miss something ?
the link you sent me doesn't include any forks

**Comment by dblock** [comment]
@aetter 👍  Want to open corresponding tickets in various repos?

**Comment by aetter** [comment]
> @aetter 👍 Want to open corresponding tickets in various repos?

Sure, done:

- https://github.com/opensearch-project/opensearch-go/issues/75
- https://github.com/opensearch-project/opensearch-java/issues/55
- https://github.com/opensearch-project/opensearch-py/issues/85
- https://github.com/opensearch-project/opensearch-js/issues/187

**Comment by rawpixel-vincent** [comment]
> I think I understand the problem now, you're saying the managed service doc should be talking about how to use forked client (https://github.com/opensearch-project/opensearch-js) instead of the AWS SDK? Or is there more to this?

yes that's pretty much it, aws docs is usually where I look for information, when I saw the low level code as how we should communicate with opensearch it left me 😮 

thanks for the information, that's awesome, looking forward to have iam / roles authentication built in then client.
